### ISSUE TITLE
Cleanup Sysvar API functions

### ISSUE THREAD

**Issue by morgo** [pingcap/tidb#36639]
## Enhancement

Currently, the sysvar API has some confusing usage. The following table summarizes the functions and proposed changes:

| Current Usage  | Problems | Suggested Fix |
| ------------- | ------------- |----------|
| `func (s *SessionVars) SetStmtVar(name string, val string) error`  | Does not do validation. Exported function name is the same as variable.SetStmtVar. It's not clear which one to use?  | Rename to private function: `func (s *SessionVars) setStmtVar(name string, val string) error`. Move variable.SetStmtVar to `func (s *SessionVars) SetStmtVar(name string, value string) error` |
| `func GetSessionOrGlobalSystemVar(s *SessionVars, name string) (string, error)`  | Inconsistent usage. Why is this not part of the SessionVars struct? | Move to `func (s *SessionVars) GetSessionOrGlobalSystemVar(name string) (string, error)` |
| `func GetSessionStatesSystemVar(s *SessionVars, name string) (string, bool, error)`  | Inconsistent usage. Why is this not part of the SessionVars struct? | Move to `func (s *SessionVars) GetSessionStatesSystemVar(name string) (string, bool, error)` |
| `func GetGlobalSystemVar(s *SessionVars, name string) (string, error)`  | Inconsistent usage. Why is this not part of the SessionVars struct? | Move to `func (s *SessionVars) GetGlobalSystemVar(name string) (string, error)` |
| `func SetStmtVar(vars *SessionVars, name string, value string) error`  | Inconsistent usage. Why is this not part of the SessionVars struct? | See suggestion on first line. |
| `func (s *SessionVars) SetSystemVar(name string, val string) error` | Function does not do normalization or validation, and is used incorrectly in several parts of the server (set value to "1" instead of "ON") | Change to do validation/normalization. The cases in the testsuite which require the old behavior can migrate to using `SetSystemVarWithRelaxedValidation` (existing function). Temporarily a function `SetSystemVarWithoutValidation` can be introduced to handle edge-cases. |
| `func SetSessionSystemVar(vars *SessionVars, name string, value string) error` | Inconsistent/duplicate usage. | This can be satisfied by `func (s *SessionVars) SetSystemVar(name string, val string) error` which is changed to do validation. |
| `func (s *SessionVars) SetSystemVarWithRelaxedValidation(name string, val string) error` | None | No change (except it might be used more since it adopts use-cases from SetSystemVar now doing validation) |

So in summary the major changes are:
- The helper functions are unified to be on the sessionVars struct (`sessionVars` is required for all changes anyway)
- `func (s *SessionVars) SetSystemVar()` now does normalization/validation.
- Overall reduction of exported functions: 2 (once SetSystemVarWithoutValidation is removed)
